docs/sources/developers/plugins/error-handling.md

```diff
@@ -0,0 +1,59 @@
++++
+title = "Error handling"
+type = "docs"
++++
+
+# Error handling
+
+This guide explains how to handle errors in plugins.
+
+## Provide usable defaults
+
+Allow the user to learn your plugin in small steps. By providing a usable default configuration, the user can get started right away, and you can avoid unnecessary error messages early on.
+
+For example, by selecting the first field of a expected type, the panel can display a visualization without any user configuration. If a user explicitly selects a field we use that one, otherwise we default to the first field of type `string` that we find:
+
+```ts
+const numberField = frame.fields.find(field =>
+  options.numberFieldName ? field.name === options.numberFieldName : field.type === FieldType.number
+);
+```
+
+## Display errors messages
+
+To display an error message to the user, `throw` an `Error` with the message you want to display:
+
+```ts
+throw new Error('An error occurred.');
+```
+
+Grafana displays the error message in the top-left corner of the panel.
+
+{{< docs-imagebox img="/img/docs/panel_error.png" class="docs-image--no-shadow" max-width="850px" >}}
+
+Avoid displaying overly technical error messages to the user. If you want to let technical users to be able to report the error, consider logging the error instead.
+
+```ts
+try {
+  failingFunction();
+} catch (err) {
+  console.error(err);
+  throw new Error('Something went wrong.');
```

Comment:
Should we really end exception messages with a period? It's normally not good style.

Reply:
It's not, but the string is shown in the UI as a error message to the user, and I think UI errors should be correct sentences.

Reply:
I guess we should add this as a note though.

Reply:
@marcusolsson it should be fixed in the UI, instead of requiring plugin authors to work around the issue.

Reply:
I agree 100%: Error handling needs to be improved for plugin development. However, this guide isn't proposing a change to how plugins handle errors, but merely documents the _current_ "best" practice. 

Regardless, after talking with UX, single sentence error messages shouldn't end with a period even in the UI, so removing the period.

Reply:
So the UI don't punctuate the error messages for a reason? That's actually the perfect example of why error/exception strings shouldn't be punctuated, because the punctuation is entirely up to whoever displays the messages (UI is only one potential destination, typically errors are also output in logs f.ex.). Glad it's resolved.